### ISSUE TITLE
How to run the container as a specific user ?

### ISSUE THREAD

**Issue by pokyah** [OmicsWorkflows/KNIME_docker_vnc#12]
First of all, thanks for this containerized working environment. This is really close to what I was looking for my team. 

I have been playing around with your dockerfiles and scripts and have pmanaged to build a functional image. 

There is a little last thing that I would like to enable : allow the container to run with the same username as the host machine.

So far I've succeeded to achieve this by  creating the user during the build stage (in the dockerfile). But this is somehow restrictive as it require to maintain a separate build for each user.

I wonder if it would be possible to create the user at the container invocation by tweaking the `entrypoint.sh` script thanks to an environment variable like SETUSERNAME that I pass at the `docker run`.  But I cannot do it with any success. 

Any hint about it?

Thanks

**Comment by weiclav** [comment]
Hi,

we tried to build docker image for security reasons by installing as many things as possible by the knimeuser, not root user. Including installation of python and R environment, and KNIME into the knimeuser home directory. We have also some Gnome related hacks (like content and settings of the panels) set by some predefined content in the defined user home directory. So, there is quite a lot of things related to this fixed knimeuser account at the moment. But I can imagine, one can install everything during the docker image build under the root to some common place and to initiate the new user on the container start using the `entrypoint.sh` script. I just have no experience with it, sorry.

We are now running the docker images on Rancher platform with not allowed root anyway. So, the image has to have some non root user already.

Take care!

**Comment by pokyah** [comment]
Hi,

Thanks for your quick reply. 

I finally found a solution.

I build the whole image with the knimuser that I previously added to the sudoers.

In `entrypoint.sh`, I use : 

```
# change the username to match host username thanks to ENV VAR SETUSERNAME passed in docker run from ./scripts
/start_container.bat
sudo usermod -l ${SETUSERNAME} knimeuser
```

And in start_container.sh, I use:

```
docker run -it -d --rm --shm-size=1g --name benworkstation%port% -p %port%:5901 -v %workspace_path%:%volume_mount_point% -e SETUSERNAME=%USERNAME% -e CONTAINER_TIMEZONE=%timezone% -e TZ=%timezone% -e VNCPASSWORD=$default_vnc_password --rm %image_name%
````

This allow me to easily rename knimeuser to our whishes (the username of the host machine).

May I askyou why did you choose to install knime in the home folder?

Best regards

**Comment by weiclav** [comment]
Hi,

thanks for your workaround details, happy to hear it has worked for you!

Regarding the reason why we put the KNIME folder into the home folder - it is nothing crucial, we got inspiration from another docker file we have used as sort of starting point where it was done this way. We have later on found out relatively handy due to quicker access to the knime.ini file if needed for some adjustments like RAM limit.

Take care
